code/modules/client/verbs/suicide.dm

```diff
@@ -22,6 +21,10 @@
 					suiciding = FALSE
 					SendSignal(COMSIG_ADD_MOOD_EVENT, "shameful_suicide", /datum/mood_event/shameful_suicide)
 					return
+
+				if(mind)
```

Comment:
is this check needed? a mindless mob would never be able to do the confirmation.

Reply:
Same is true for other checks I guess.